pkg/controller/windowsmachineconfig/windowsmachineconfig_controller.go

```diff
@@ -166,6 +202,36 @@ func (r *ReconcileWindowsMachineConfig) getCloudProvider(instance *wmcapi.Window
 // Result.Requeue is true, otherwise upon completion it will remove the work from the queue.
 func (r *ReconcileWindowsMachineConfig) Reconcile(request reconcile.Request) (reconcile.Result, error) {
 	log.Info("reconciling", "namespace", request.Namespace, "name", request.Name)
+	wmcoObject := false
+	// Fetch the Machine instance
+	Machine := &mapiv1.Machine{}
+	provisionedPhase := "Provisioned"
+	if err := r.client.Get(context.TODO(), request.NamespacedName, Machine); err != nil {
+		wmcoObject = true
+		// TODO: Uncomment this when we switch to Machine-api completely
+		//if k8sapierrors.IsNotFound(err) {
+		//	// Request object not found, could have been deleted after reconcile request.
+		//	// Owned objects are automatically garbage collected. For additional cleanup logic use finalizers.
+		//	// Return and don't requeue
+		//
+		//	return reconcile.Result{}, nil
+		//
+		//}
+		//// Error reading the object - requeue the request.
+		//return reconcile.Result{}, err
+	}
+	if !wmcoObject {
+		if Machine.Status.Phase == nil {
+			// Phase can be nil and should be ignored by WMCO
+			return reconcile.Result{}, nil
+		}
+		if *Machine.Status.Phase != provisionedPhase {
+			// If the Machine is not in provisioned state, WMCO shouldn't care about it
+			return reconcile.Result{}, nil
+		}
+		r.recorder.Eventf(Machine, corev1.EventTypeNormal, "WMCO Update",
+			"Machine %s Provisioned Successfully", Machine.Name)
+	}
```

Comment:
Shouldn't there be a return here?

Reply:
Doesn't matter but it won't hurt to have it here.